File: aula12/Aula12.java
public class Main {

    public static void main(String[] args) {

    Insumo insumo = new Insumo("Amendoim", 50, 8);

    insumo.exibirDados();

    int mesAtual = 7;

    if (insumo.estaValido(mesAtual)) {
        System.out.println("Produto válido.");
    } else {
        System.out.println("Produto vencido.");
    }
}
}

class Insumo {

    private String nome;
    private int quantidade;
    private int mesValidade;

    public Insumo(String nome, int quantidade, int mesValidade) {
        this.nome = nome;
        this.quantidade = quantidade;
        this.mesValidade = 1; 
        setMesValidade(mesValidade);
    }

    public void setMesValidade(int mes) {
        if (mes >= 1 && mes <= 12) {
            this.mesValidade = mes;
        } else {
            System.out.println("ALERTA: Mês inválido! Digite um valor entre 1 e 12.");
        }
    }

    public boolean estaValido(int mesAtual) {
        if (mesValidade < mesAtual) {
            System.out.println("ALERTA: Risco de Amendoim Murcho! Produto Vencido");
            return false;
        }
        return true;
    }

    public void exibirDados() {
        System.out.println("Nome: " + nome);
        System.out.println("Quantidade: " + quantidade);
        System.out.println("Mês de validade: " + mesValidade);
    }
}

File: aula12/Aula12.py
class Insumo:
    def __init__(self, nome, quantidade, mes_validade):
        self.__nome = nome
        self.__quantidade = quantidade
        self.__mes_validade = 1  # valor inicial seguro
        self.set_mes_validade(mes_validade)

    def set_mes_validade(self, mes):
        if 1 <= mes <= 12:
            self.__mes_validade = mes
        else:
            print("ALERTA: Mês inválido! Digite um valor entre 1 e 12.")

    def esta_valido(self, mes_atual):
        if self.__mes_validade < mes_atual:
            print("ALERTA: Risco de Amendoim Murcho! Produto Vencido")
            return False
        return True

    def exibir_dados(self):
        print(f"Nome: {self.__nome}")
        print(f"Quantidade: {self.__quantidade}")
        print(f"Mês de validade: {self.__mes_validade}")

insumo = Insumo("Amendoim", 50, 8)

insumo.exibir_dados()

mes_atual = 7

if insumo.esta_valido(mes_atual):
    print("Produto válido.")
else:
    print("Produto vencido.")